### ISSUE TITLE
Serilog Implementation - Hijack ILoggerFactory

### ISSUE THREAD

**Issue by DOMZE** [SteeltoeOSS/steeltoe#4]
Hello,

I first off want to thank you guys for the great work of the steeltoe library for .NET, really is great.

I wanted to see with you guys if you were open to have a UseSerilogDynamicConsole implementation a bit like UseSerilog.

The problem I had is that I wanted to replace the whole console logger with the one provided by Serilog. I also wanted to control my loggerconfiguration through code rather than through the configuration file (appsettings) as certain sinks are tough to control using the json settings configuration.

I can make a PR with the changes if you wish. I tested it and it works the way i want it to work.
Basically it consists of having an extension on IWebHostBuilder, having a Factory class (similar to Serilog) and using your dynamicprovider (using another constructor)

Let me know if it's something you guys would be interested in having. It definitely can be enhanced to mimick UseSerilog extension, but I wanted to make it work for my needs.

**Comment by hananiel** [comment]
We need to pull this one in before closing ... https://github.com/SteeltoeOSS/Logging/pull/16

**Comment by hananiel** [comment]
@DOMZE Sorry this took me so long to get to. I am trying to pull your changes in but have a few questions. Can you give the example of how you are using it  - an example of invoking UseSerilogDynamicConsole((builderContext, loggerConfiguration=>{ ...what are you doing here?... }). I am wondering if I can reduce this to one more flexible extension vs having two ways one more flexible than other.

**Comment by DOMZE** [comment]
Hi @hananiel 

Yes, the idea is that I have a Http sink that I want to configure the endpoints and api keys for instance (through a custom http client) and the configuration comes from the IConfiguration (hostingContext.Configuration)

An example can be
`var httpLoggingClient = new HttpLoggingClient(hostingContext.Configuration["ApiKey"], hostingContext.Configuration["ClientId"]);
loggerConfiguration
	.MinimumLevel.Information()
	.Enrich.WithExceptionDetails()
	.MinimumLevel.Override("Microsoft", LogEventLevel.Warning)
	.WriteTo.Http($"{hostingContext.Configuration["LoggingEndPoint"]}", httpClient: httpLoggingClient, textFormatter: new CompactJsonFormatter());`

**Comment by hananiel** [comment]
@DOMZE thats great. Thank you for the sample
